src/redux/books/books.js

```diff
@@ -1,34 +1,69 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+const GET_BOOK = 'bookstore/books/GET_BOOK';
 
 const prevState = [];
-
-// Action Logic
-const addBook = (book) => ({
-  type: ADD_BOOK,
-  book,
-});
-
-const removeBook = (id) => ({
-  type: REMOVE_BOOK,
-  id,
-});
+const source = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/BoLpCLs90yg1VqJcibSz/books';
 
 // Book Reducer logic
 const BooksRedux = (state = prevState, action) => {
   switch (action.type) {
+    case GET_BOOK:
+      return action.books;
+
     case ADD_BOOK:
       return [
         ...state,
         action.book,
       ];
     case REMOVE_BOOK:
-      return [
-        ...state.filter((book) => book.id !== action.id),
-      ];
+      return state.filter((book) => book.item_id !== action.itemB_id);
 
     default: return state;
   }
 };
-export { addBook, removeBook };
+
+/* Actions Logic */
+
+// Fetch Books
+const getBooks = createAsyncThunk(GET_BOOK, async (post, { dispatch }) => {
+  const res = await fetch(source);
+  const data = await res.json();
+  const books = Object.keys(data).map((key) => ({
+    ...data[key][0],
+    item_id: key,
+  }));
+  dispatch({
+    type: GET_BOOK,
+    books,
+  });
+});
```

Comment:
- To make your code cleaner, I suggest you implement an action creator `fetchBook` just like you did in the previous milestone, and dispatch this action here. Please fix this 🙏

Reply:
Hello, @mckent05 
Thank you for your time and the informative feedback. I am a bit confused about what you intend that I should do. I was about asking if it is okay to call you. I am sorry to take your time please.